cvc5_pythonic_api/cvc5_pythonic.py

```diff
@@ -6202,6 +6202,35 @@ def model(self):
         """
         return ModelRef(self)
 
+    def proof(self):
+        """Return a proof for the last `check()`.
+
+        This function raises an exception if
+        a proof is not available (e.g., last `check()` does not return unsat).
+
+        >>> s = Solver()
+        >>> s.set('produce-proofs','true')
+        >>> a = Int('a')
+        >>> s.add(a + 2 == 0)
+        >>> s.add(a == 0)
+        >>> s.check()
+        unsat
+        >>> s.proof()
+        (SCOPE: Not(And(a + 2 == 0, a == 0)),
+         (SCOPE: Not(And(a + 2 == 0, a == 0)),
+          [a + 2 == 0, a == 0],
+          (EQ_RESOLVE: False,
+           (ASSUME: a == 0, [a == 0]),
+           (MACRO_SR_EQ_INTRO: (a == 0) == False,
+            [a == 0, 7, 12],
+            (EQ_RESOLVE: a == -2,
+             (ASSUME: a + 2 == 0, [a + 2 == 0]),
+             (MACRO_SR_EQ_INTRO: (a + 2 == 0) == (a == -2),
+              [a + 2 == 0, 7, 12]))))))
+        """
```

Comment:
Add a test for the exception, like [here](https://github.com/cvc5/cvc5_pythonic_api/blob/main/cvc5_pythonic_api/cvc5_pythonic.py#L40).